### PR TITLE
vg/vgsvg: drop fontMap, use informations from font.Font

### PR DESCRIPTION
**Pull request by sbinet**
This CL drops the use of the internal fontMap that was associating some
pre-defined set of fonts with a set of SVG naming schemes (derived from
PostScript).
Instead, use the informations contained in plot/font.Font to derive the
expected SVG font-family (and friends) font style string.

Updates #702.

Please take a look.

### CODE REVIEW

**Comment by sbinet** [comment]
PTAL @stippi2

File: vg/vgsvg/vgsvg.go [hunk 1]
```diff
@@ -345,38 +351,87 @@ func (c *Canvas) DrawImage(rect vg.Rectangle, img image.Image) {
...
+	// https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/font-style
+	switch sty {
+	case stdfnt.StyleNormal:
+		return "normal"
+	case stdfnt.StyleItalic:
+		return "italic"
+	case stdfnt.StyleOblique:
+		return "oblique"
+	}
+	panic(fmt.Errorf("vgsvg: invalid font style %+v (v=%d)", sty, int(sty)))
```
Comment (kortschak) [inline vg/vgsvg/vgsvg.go:406]: default?

File: vg/vgsvg/vgsvg.go [hunk 2]
```diff
@@ -345,38 +351,87 @@ func (c *Canvas) DrawImage(rect vg.Rectangle, img image.Image) {
...
+	case stdfnt.WeightSemiBold:
+		return "600"
+	case stdfnt.WeightBold:
+		return "bold"
+	case stdfnt.WeightExtraBold:
+		return "800"
+	case stdfnt.WeightBlack:
+		return "900"
+	}
+	panic(fmt.Errorf("vgsvg: invalid font weight %+v (v=%d)", w, int(w)))
```
Comment (kortschak) [inline vg/vgsvg/vgsvg.go:433]: default?

**Review by kortschak — COMMENTED** [review]
This looks like it has broken text alignment; the width of text in the test golden values is greater and so now the text is not centred and extends beyond the bounds of the plot.

**Comment by sbinet** [comment]
@kortschak what do you see?

here is what I get:
screenshot

could it be that the Liberation fonts are not available to your browser?

**Comment by kortschak** [comment]
Screenshot from 2021-06-15 07-11-45

**Comment by sbinet** [comment]
I get this with my browser:
screenshot

(`master` is left.)

and this in the github rich-diff:
screenshot

**Comment by kortschak** [comment]
It's entirely possible that the installed fonts impacts on this. There should probably be some documentation explaining how to ensure that fonts are available for SVG-rendered plots with this change. We're bound to get issues otherwise.

**Comment by sbinet** [comment]
you're probably right.

perhaps this could be addressed with embedding fonts?
_i.e.:_ https://github.com/gonum/plot/issues/703 and adding the proper documentation+example.

**Comment by sbinet** [comment]
@kortschak 
I have a CL ready to add fonts embedding to SVG canvases.
should I add it to this PR or send another one (once that one is merged.)

**Comment by kortschak** [comment]
Are they independent? If they are, then looking at the PR adding embedding and then rebasing this only that once merged would be the best I think.

**Comment by sbinet** [comment]
they are independent.
I'll send a PR.

**Comment by sbinet** [comment]
actually, it's a bit more intertwined than I remembered. (the coupling is in the naming taxonomy of fonts)
it's #705.

**Comment by sbinet** [comment]
superseded by https://github.com/gonum/plot/pull/705

**Pull request closed without merging**